<?xml version="1.0" ?>

<robot name="kuka">
	<link name="base_link">
		<visual>
			<geometry>
				<cylinder length="0.2" radius="0.15"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.1"/>
			<material name="grey">
				<color rgba="0.5 0.5 0.5 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.2" radius="0.15"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.1"/>
		</collision>	
	</link>

  	
	<joint name="first_joint" type="revolute">
		<parent link="base_link"/>
		<child link="first_link"/>
		<origin xyz="0 0 0.1"/>
		<axis xyz="0 0 1"/>
		<limit effort="300" velocity="0.1" lower="-3.14" upper="3.14"/>
	</joint>

	
	<link name="first_link">
		<visual>
			<geometry>
				<cylinder length="0.3" radius="0.1"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.15"/>
			<material name="blue">
				<color rgba="0 0 1 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.3" radius="0.1"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.15"/>
		</collision>		
		<inertial>
			<mass value="3"/>
			<inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
		</inertial>
	</link>

  <joint name="second_joint" type="revolute">
		<parent link="first_link"/>
		<child link="second_link"/>
		<origin xyz="0 0.1 0.3"/>
		<axis xyz="0 1 0"/>
		<limit effort="300" velocity="0.1" lower="-0.6" upper="0.6"/>
	</joint>

  <link name="second_link">
		<visual>
			<geometry>
				<cylinder length="0.3" radius="0.1"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0.1 0.15"/>
			<material name="red">
				<color rgba="1 0 0 1"/>
			</material>
		</visual>
		<collision>
			<geometry>
				<cylinder length="0.3" radius="0.11"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0.1 0.15"/>
		</collision>
		<inertial>
			<mass value="3"/>
			<inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
		</inertial>
	</link>

  <joint name="third_joint" type="revolute">
		<parent link="second_link"/>
		<child link="third_link"/>
		<origin xyz="0 0.1 0.3"/>
		<axis xyz="0 0 1"/>
		<limit effort="300" velocity="0.1" lower="-3.14" upper="3.14"/>
	</joint>

  	
	<link name="third_link">
		<visual>
			<geometry>
				<cylinder length="0.3" radius="0.1"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.15"/>
			<material name="blue">
				<color rgba="0 0 1 1"/>
			</material>
		</visual>
		
		<collision>
			<geometry>
				<cylinder length="0.3" radius="0.11"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.15"/>
		</collision>
		
		<inertial>
			<mass value="3"/>
			<inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
		</inertial>
	</link>

  <joint name="fourth_joint" type="revolute">
		<parent link="third_link"/>
		<child link="fourth_link"/>
		<origin xyz="0 -0.1 0.3"/>
		<axis xyz="0 1 0"/>
		<limit effort="300" velocity="0.1" lower="-3.14" upper="3.14"/>
	</joint>

  <link name="fourth_link">
		<visual>
			<geometry>
				<cylinder length="0.3" radius="0.1"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 -0.1 0.15"/>
			<material name="red">
				<color rgba="1 0 0 1"/>
			</material>
		</visual>
		
		<collision>
			<geometry>
				<cylinder length="0.3" radius="0.11"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 -0.1 0.15"/>
		</collision>
		
		<inertial>
			<mass value="3"/>
			<inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
		</inertial>
	</link>

  <joint name="fifth_joint" type="revolute">
		<parent link="fourth_link"/>
		<child link="fifth_link"/>
		<origin xyz="0 -0.1 0.3"/>
		<axis xyz="0 0 1"/>
		<limit effort="300" velocity="0.1" lower="-3.14" upper="3.14"/>
	</joint>

  <link name="fifth_link">
		<visual>
			<geometry>
				<cylinder length="0.3" radius="0.1"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.15"/>
			<material name="blue">
				<color rgba="0 0 1 1"/>
			</material>
		</visual>
		
		<collision>
			<geometry>
				<cylinder length="0.3" radius="0.11"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.15"/>
		</collision>
		
		<inertial>
			<mass value="3"/>
			<inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
		</inertial>
	</link>

  <joint name="sixth_joint" type="revolute">
		<parent link="fifth_link"/>
		<child link="sixth_link"/>
		<origin xyz="0 0.1 0.3"/>
		<axis xyz="0 1 0"/>
		<limit effort="300" velocity="0.1" lower="-3.14" upper="3.14"/>
	</joint>

  <link name="sixth_link">
		<visual>
			<geometry>
				<cylinder length="0.3" radius="0.1"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0.1 0.15"/>
			<material name="red">
				<color rgba="1 0 0 1"/>
			</material>
		</visual>
		
		<collision>
			<geometry>
				<cylinder length="0.3" radius="0.11"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0.1 0.15"/>
		</collision>
		
		<inertial>
			<mass value="3"/>
			<inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
		</inertial>
	</link>

  <joint name="seventh_joint" type="revolute">
		<parent link="sixth_link"/>
		<child link="seventh_link"/>
		<origin xyz="0 0.1 0.3"/>
		<axis xyz="0 0 1"/>
		<limit effort="300" velocity="0.1" lower="-3.14" upper="3.14"/>
	</joint>

  <link name="seventh_link">
		<visual>
			<geometry>
				<cylinder length="0.1" radius="0.05"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.05"/>
			<material name="blue">
				<color rgba="0 0 1 1"/>
			</material>
		</visual>
		
		<collision>
			<geometry>
				<cylinder length="0.1" radius="0.06"/>
			</geometry>
			<origin rpy="0 0 0" xyz="0 0 0.05"/>
		</collision>
		
		<inertial>
			<mass value="1"/>
			<inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0"/>
		</inertial>
	</link>

</robot>
